pkg/vendir/fetch/image/imgpkg.go

```diff
@@ -135,3 +238,29 @@ func (t *Imgpkg) addDangerousArgs(args []string) []string {
 	}
 	return args
 }
+
+// Logger provided to the imgpkg API calls
+type Logger struct {
```

Comment:
ok i'll admit,  i'm confused why, in a PR about caching, there's a logger that looks like it has levels but actually they're hardcoded and you'll never know which level something was logged at, especially if it was a debug log bc those don't even log at all? 

My guess is you needed an interface to pass to something else, but maybe some more elaborate hint in the comments would be helpful

Reply:
i guess in general --  i like log messages.  like being able to get debug output.  i have some dreams w.r.t . being able to trigger all the tools in kapp-controller into debug at the same time, for the sake of figuring out what's going on.

Reply:
The logger created in this particular instance is a bit peculiar because it needs to conform to the requirements from the imgpkg API.
Because of that, it is only present in the `image` package. In the future when we are using our tools and can provide directives to activate different logging levels we should definitely revisit all these instances to ensure that we are doing the correct thing.
In the meanwhile, I will update the logger struct comment to inform the user that not all the logs will be present